File: solutions/python/kindergarten-garden/2/kindergarten_garden.py
STUDENTS = [
    "Alice",
    "Bob",
    "Charlie",
    "David",
    "Eve",
    "Fred",
    "Ginny",
    "Harriet",
    "Ileana",
    "Joseph",
    "Kincaid",
    "Larry"
]

PLANTS = {
    "G": "Grass",
    "C": "Clover",
    "R": "Radishes",
    "V": "Violets"
}

class Garden:
    def __init__(self, diagram, students=STUDENTS):
        self.diagram = diagram.split("\n")
        self.students = sorted(students)

    def plants(self, student):
        student_index = self.students.index(student) * 2
        return [PLANTS[p] for p in self.diagram[0][student_index:student_index+2] + self.diagram[1][student_index:student_index+2]]

File: solutions/csharp/kindergarten-garden/1/KindergartenGarden.cs
public enum Plant {
  Violets,
  Radishes,
  Clover,
  Grass
}

public enum Student {
  Alice,
  Bob,
  Charlie,
  David,
  Eve,
  Fred,
  Ginny,
  Harriet,
  Ileana,
  Joseph,
  Kincaid,
  Larry
}

public class KindergartenGarden {
  private readonly Dictionary<char, Plant> _plants = new() {
    {'V', Plant.Violets},
    {'R', Plant.Radishes},
    {'C', Plant.Clover},
    {'G', Plant.Grass}
  };

  private readonly Dictionary<Student, List<Plant>> _diagram = new() {
    {Student.Alice, new()},
    {Student.Bob, new()},
    {Student.Charlie, new()},
    {Student.David, new()},
    {Student.Eve, new()},
    {Student.Fred, new()},
    {Student.Ginny, new()},
    {Student.Harriet, new()},
    {Student.Ileana, new()},
    {Student.Joseph, new()},
    {Student.Kincaid, new()},
    {Student.Larry, new()}
  };

  public KindergartenGarden(string diagram) {
    var lines = diagram.Split("\n");
    var studentIndex = 0;
    for (var i = 0; i < lines[0].Count() - 1; i += 2) {
      _diagram[(Student)studentIndex].Add(_plants[lines[0][i]]);
      _diagram[(Student)studentIndex].Add(_plants[lines[0][i + 1]]);
      _diagram[(Student)studentIndex].Add(_plants[lines[1][i]]);
      _diagram[(Student)studentIndex].Add(_plants[lines[1][i + 1]]);
      studentIndex++;
    }
  }

  public IEnumerable<Plant> Plants(string student) {
    return _diagram[Enum.Parse<Student>(student)].ToArray();
  }
}